### DISCUSSION TITLE
C++ standard

### DISCUSSION THREAD

**Discussion by jmirabel** [stack-of-tasks/pinocchio#1361] [General]
I wonder what prevents Pinocchio to switch to C++ 11.

**Comment by jcarpent** [comment]
It is planned for Pinocchio 3x, but for Pinocchio 2.x, it would be nice to stick with C++98.
What do you expect from C++11 in particular?

**Reply by jmirabel** [reply]
I see mostly too advantages : possible simplification of the code (or at least, make new code easier to write) and get rid of some boost module.

As a side note, one needs C++17 to replace `boost::variant` by `std::variant`.

**Reply by cmastalli** [reply]
Should we get rid of some boost module? If so, which modules have computational disadvantages with respect to std library?

In case that the computational performance is the same, then what are the benefits of partially migrate some of the modules if we still will depend on Boost (e.g. Boost Python)?

**Reply by jcarpent** [reply]
@jmirabel we discussed already several times but std variant are less advanced than boost ones.

**Reply by jmirabel** [reply]
I wasn't aware of the discussion you mention and did not find anything relevant when searching for related issues. Of course, I may have forgotten a discussion or missed a relevant issue.

Anyway, my point was not to put stress on the timeline but rather to know what to expect for future releases of Pinocchio.

**Comment by gabrielebndn** [comment]
I guess mostly compatibility for external users with C++98? Don't know how many people out there still use C++98. IMHO, it seems to me a good plan to wait for a change in major version to drop compatibility with C++98

**Comment by jcarpent** [comment]
Currently, I don’t see obvious reason to force the migration. We can still perform template generation for C++11 users without big change. At the time being, I have on my side other priorities to enhance Pinocchio.
